server/src/internalClusterTest/java/org/opensearch/search/nested/SimpleNestedIT.java

```diff
@@ -500,6 +499,98 @@ public void testExplain() throws Exception {
         assertThat(explanation.toString(), startsWith("0.36464313 = Score based on 2 child docs in range from 0 to 1"));
     }
 
+    public void testExplainMultipleDocIndexingNoDeletedDoc() throws Exception {
+        assertAcked(
+            prepareCreate("test").setMapping(
+                jsonBuilder().startObject()
+                    .startObject("properties")
+                    .startObject("nested1")
+                    .field("type", "nested")
+                    .endObject()
+                    .endObject()
+                    .endObject()
+            )
+        );
+
+        ensureGreen();
+
+        for (int i = 0; i < 5; i++) {
+            client().prepareIndex("test")
+                .setId(String.valueOf(i))
+                .setSource(
+                    jsonBuilder().startObject()
+                        .field("field" + i, "value" + i)
+                        .startArray("nested" + i)
+                        .startObject()
+                        .field("n_field" + i, "n_value" + i)
+                        .endObject()
+                        .startObject()
+                        .field("n_field" + i, "n_value" + i)
+                        .endObject()
+                        .endArray()
+                        .endObject()
+                )
+                .setRefreshPolicy(IMMEDIATE)
+                .get();
+        }
+
+        SearchResponse searchResponse = client().prepareSearch("test")
+            .setQuery(nestedQuery("nested1", termQuery("nested1.n_field1", "n_value1"), ScoreMode.Total))
+            .setExplain(true)
+            .get();
+        assertNoFailures(searchResponse);
+        assertThat(searchResponse.getHits().getTotalHits().value, equalTo(1L));
+        Explanation explanation = searchResponse.getHits().getHits()[0].getExplanation();
+        assertThat(explanation.getValue(), equalTo(searchResponse.getHits().getHits()[0].getScore()));
+        assertThat(explanation.toString(), startsWith("0.36464313 = Score based on 2 child docs in range"));
```

Comment:
This may be slightly outside of the scope of this test but just thinking out loud -- If we know the number of deleted docs is it possible to also know the specific range?

Reply:
Not sure on how to calculate range on deleted doc count basis, from the code it doesnt really look that we can directly calculate range from deleted doc count: https://github.com/apache/lucene/blob/a6f70ad2bb0b682eb65feb522358ee6d16cad766/lucene/join/src/java/org/apache/lucene/search/join/ToParentBlockJoinQuery.java#L402


To make the difference significant in the tests, I have added a separate assertion for range for `testExplainSingleDocIndexingNoDeletedDoc` which is non-existent in the other tests for explain.